app/controllers/schools/advice/base_out_of_hours_controller.rb

```diff
@@ -15,12 +14,19 @@ def insights
       def analysis
         @annual_usage_breakdown = annual_usage_breakdown_service.usage_breakdown
         @holiday_usage = holiday_usage_calculation_service.school_holiday_calendar_comparison
-        @analysis_dates = analysis_dates
         @meter_selection = Charts::MeterSelection.new(@school, aggregate_school, advice_page_fuel_type, date_window: 363)
       end
 
       private
 
+      def check_can_run_analysis
+        @analysable = create_analysable
```

Comment:
would it be the same/clearer to call super for the first 2 lines?

Reply:
The `render ... and return` doesn't seem to work correctly when calling super as it returns to the overriding method.